### ISSUE TITLE
ags4 new compiler string on struct error

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#1312]
**Describe the bug**
I can't store a string in a non-managed struct when using the new compiler, it gives me no errors, but once the game loads I get the following error:

Error: internal error: MEMWRITEPTR argument is not dynamic object

**AGS Version**
AGS 4, [98b56ebb7ae722a6d387aa88c509a281382994f1](https://github.com/adventuregamestudio/ags/tree/98b56ebb7ae722a6d387aa88c509a281382994f1) | Cirrus artifact: [task/5983374398455808](https://cirrus-ci.com/task/5983374398455808)

**Game**
[compilerbugags4.zip](https://github.com/adventuregamestudio/ags/files/6635244/compilerbugags4.zip)

**To Reproduce**
1. Select the new script compiler
2. Put the following code in a script module:
```AGS Script
struct Stru {
  String Txt;
};

void game_start()
{
  Stru a;
  a.Txt = "Cause bug!";
}
```
4. Build, run and see error

**Expected behavior**
It should not cause error.

**Screenshots**
![image](https://user-images.githubusercontent.com/2244442/121618546-556db100-ca3d-11eb-8936-c8efe68593cc.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 10

**Comment by ivan-mogilko** [comment]
@ericoporto not related to the bug, but will it be easy to add new compiler tests to CI too? (#1217)

**Comment by ericoporto** [comment]
Ah, it is easy, yes! I didn't saw that issue. I will add.

**Comment by fernewelten** [comment]
> Error: internal error: MEMWRITEPTR argument is not dynamic object

I'm having a look at it ...

**Comment by ericoporto** [comment]
The PR fixed for me, but as requested I opened a PR (#1314) to add tests for the compiler and found a failing test.

**Comment by fernewelten** [comment]
> found a failing test.

Addressed in #1315.

**Comment by ivan-mogilko** [comment]
This ticket is resolved by #1313.
